src/main/drivers/rcdevice.c

```diff
@@ -0,0 +1,1143 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "common/crc.h"
+#include "common/maths.h"
+#include "drivers/time.h"
+#include "rcdevice.h"
+
+#include "config/feature.h"
+#include "fc/config.h"
```

Comment:
Device drivers should not call "upward", that is they should not make calls to thing above the device driver layer. So you should not include anything from `fc` or `io`.

So remove `#include "fc/config.h"` and move any dependent code elsewhere.

I think you can probably replace `#include "io/serial.h"` by `#include "drivers/serial.h"` - if not, again remove dependent code elsewhere.

Also remove `#include "io/rcdevice_cam.h"` and any dependent code.

Reply:
Understand, I'm fixing the problems that you guys post.  Thank you!